configuration/azure/appconfig/appconfig.go

```diff
@@ -248,9 +266,103 @@ func (r *ConfigurationStore) getLabelFromMetadata(metadata map[string]string) *s
 }
 
 func (r *ConfigurationStore) Subscribe(ctx context.Context, req *configuration.SubscribeRequest, handler configuration.UpdateHandler) (string, error) {
-	return "", fmt.Errorf("Subscribe is not implemented by this configuration store")
+	subscribeID := uuid.New().String()
+	stop := make(chan struct{})
+	r.subscribeStopChanMap.Store(subscribeID, stop)
+	sentinelKey := r.getSentinelKeyFromMetadata(req.Metadata)
+	if sentinelKey == "" {
+		return "", fmt.Errorf("sentinel key is not provided in metadata")
+	}
+	actualKeyToSubscribeSlice := make([]string, 0)
+	for _, key := range req.Keys {
+		_, found := r.keysToSubIDMap[key]
+		if !found {
+			actualKeyToSubscribeSlice = append(actualKeyToSubscribeSlice, key)
+			r.keysToSubIDMap[key] = subscribeID
+		}
+	}
+	if len(actualKeyToSubscribeSlice) == 0 {
+		return "", fmt.Errorf("all provided keys are already subscribed")
+	}
+	value, found := r.subIDToKeyListMap[subscribeID]
+	if found {
+		value = append(value, actualKeyToSubscribeSlice...)
+		r.subIDToKeyListMap[subscribeID] = value
+	} else {
+		r.subIDToKeyListMap[subscribeID] = actualKeyToSubscribeSlice
+	}
+
+	req.Keys = actualKeyToSubscribeSlice
+	go r.doSubscribe(ctx, req, handler, sentinelKey, subscribeID, stop)
+	return subscribeID, nil
+}
+
+func (r *ConfigurationStore) doSubscribe(ctx context.Context, req *configuration.SubscribeRequest, handler configuration.UpdateHandler, sentinelKey string, id string, stop chan struct{}) {
+	for {
+		// get sentinel key changes
+		_, err := r.Get(ctx, &configuration.GetRequest{
+			Keys:     []string{sentinelKey},
+			Metadata: req.Metadata,
+		})
+		if err != nil {
+			r.logger.Debugf("fail to get sentinel key changes or sentinel key's value is unchanged: %s", err)
+		} else {
+			items, err := r.Get(ctx, &configuration.GetRequest{
+				Keys:     req.Keys,
+				Metadata: req.Metadata,
+			})
+			if err != nil {
+				r.logger.Errorf("fail to get configuration key changes: %s", err)
+			} else {
+				r.handleSubscribedChange(ctx, req, handler, items, id)
+			}
+		}
+		select {
+		case <-stop:
+			return
+		case <-ctx.Done():
+			return
+		case <-time.After(r.metadata.subscribePollInterval):
```

Comment:
Just to confirm, the next iteration of for will go through only after poll interval + handler block time have passed correct? So the poll interval is in addition to how long the app takes to handle a particular key update....

Reply:
yes